nautobot/ipam/models.py

```diff
@@ -600,6 +620,34 @@ def save(self, *args, **kwargs):
             parent = max(supernets, key=operator.attrgetter("prefix_length"))
             self.parent = parent
 
+        # Validate that creation of this prefix does not create an invalid parent/child relationship
+        if self.parent and self.parent.type != constants.PREFIX_ALLOWED_PARENT_TYPES[self.type]:
+            err_msg = f"{self.type.title()} prefixes cannot be children of {self.parent.type.title()} prefixes"
```

Comment:
What if anything do we need to do in the data migrations to ensure that existing data follows these constraints? Or is that future work (which I'm fine with if so, so long as it's tracked)?

Reply:
We didn't account for these constraints when creating the migrations so we'll unfortunately need to circle back to make sure we're not assigning IP addresses to pool prefixes, make sure parent prefixes are containers and likely create new network prefixes to contain IP addresses if necessary. I opened #4004 for this work.